catalog/core/catalog-core-solrcommands/src/main/java/org/codice/ddf/commands/solr/BackupCommand.java

```diff
@@ -15,36 +15,99 @@
+                    + "backups, the backup location must be shared by all Solr nodes.")
     String backupLocation;
 
     @Option(name = "-c", aliases = {"--coreName"}, multiValued = false, required = false,
-            description = "Name of the Solr core to be backed up.  If not specified, the 'catalog' core will be backed up.")
+            description = "Name of the Solr core/collection to be backed up.  If not specified, the 'catalog' core/collection will be backed up.")
     String coreName = "catalog";
 
+    @Option(name = "-a", aliases = {"--asyncBackup"}, multiValued = false, required = false,
+            description = "Perform an asynchronous backup of Solr Cloud.")
+    boolean asyncBackup;
+
+    @Option(name = "-s", aliases = {"--asyncBackupStatus"}, multiValued = false, required = false,
+            description = "Get the status of a Solr Cloud asynchronous backup. Used in conjunction with --asyncBackupReqId.")
+    boolean asyncBackupStatus;
+
+    @Option(name = "-i", aliases = {"--asyncBackupReqId"}, multiValued = false, required = false,
+            description = "Request Id returned after performing a Solr Cloud asynchronous backup. This request Id is used to track"
+                    + " the status of a given backup. When requesting a backup status, --asyncBackupStatus and --asyncBackupReqId"
+                    + " are both required options.")
+    String asyncBackupReqId;
+
     @Option(name = "-n", aliases = {"--numToKeep"}, multiValued = false, required = false,
             description =
-                    "Number of backups to be maintained.  If this backup operation will result"
-                            + " in exceeding this threshold, the oldest backup will be deleted.")
+                    "Number of backups to be maintained.  If this backup opertion will result"
+                            + " in exceeding this threshold, the oldest backup will be deleted. This option is not supported"
+                            + " when backing up Solr Cloud.")
     int numberToKeep = 0;
 
+    private static final String SOLR_CLIENT_PROP = "solr.client";
+
+    private static final String CLOUD_SOLR_CLIENT_TYPE = "CloudSolrClient";
+
     @Override
     public Object doExecute() throws Exception {
 
+        if (isSystemConfiguredWithSolrCloud()) {
+            SolrClient client = null;
+            try {
+                client = getSolrClient();
+
+                if(client == null) {
+                    printErrorMessage(String.format(
+                            "Could not determine Zookeeper Hosts. Please verify that the system property %s is configured in %s.",
+                            ZOOKEEPER_HOSTS_PROP, Paths.get(System.getProperty("ddf.home"), "etc", "system.properties")));
+                    return null;
+                }
+
+                if (asyncBackupStatus) {
+                    if (!isCloudBackupStatusInputValid()) {
```

Comment:
❓ Would it be clearer if the method only did the check and the `if` block printed out the message? Makes the method more re-usable and people don't have to look inside the method to make sure it does print something. Or we could call the method `verifyCloudBackupStatusInputValid()` and have it print the error and throw an exception.